src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java

```diff
@@ -0,0 +1,270 @@
+import com.microsoft.azure.spring.data.cosmosdb.performance.repository.PerfPersonRepository;
+import com.microsoft.azure.spring.data.cosmosdb.performance.service.SdkService;
+import com.microsoft.azure.spring.data.cosmosdb.performance.utils.Constants;
+import com.microsoft.azure.spring.data.cosmosdb.performance.utils.DatabaseUtils;
+import com.microsoft.azure.spring.data.cosmosdb.performance.utils.PerfDataProvider;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.microsoft.azure.spring.data.cosmosdb.performance.utils.FunctionUtils.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = PerfConfiguration.class)
+public class PerformanceCompareIT {
+    @Autowired
+    private DocumentClient documentClient;
+
+    @Autowired
+    private PerfPersonRepository repository;
+
+    private static final int TIMES = 100;
+    private static final int BATCH_SIZE = 10;
+    private static final float ACCEPTANCE_DIFF = 0.1f;
+    private static boolean hasInit = false;
+    private static SdkService sdkService;
+    private static PerformanceReport report = new PerformanceReport();
+
+    @Before
+    public void setup() throws DocumentClientException {
+        if (!hasInit) {
+            DatabaseUtils.createDatabase(documentClient, Constants.PERF_DATABASE_NAME);
+            DatabaseUtils.createCollection(documentClient, Constants.PERF_DATABASE_NAME,
+                    Constants.SPRING_COLLECTION_NAME);
+            DatabaseUtils.createCollection(documentClient, Constants.PERF_DATABASE_NAME, Constants.SDK_COLLECTION_NAME);
+
+            sdkService = new SdkService(documentClient, Constants.PERF_DATABASE_NAME, Constants.SDK_COLLECTION_NAME);
+            hasInit = true;
+        }
+    }
+
+    @After
+    public void clear() {
+        repository.deleteAll();
+        sdkService.deleteAll();
+    }
+
+    @AfterClass
+    public static void printReport() {
+        report.getPerfItems().forEach(System.out::println);
+    }
+
+    @Test
+    public void saveOneRecordTest() {
+        final List<PerfPerson> personList = PerfDataProvider.getPerfData(TIMES);
+
+        final long springCost = applyInputListFunc(personList, repository::save);
+        final long sdkCost = applyInputListFunc(personList, sdkService::save);
+
+        report.addItem(new PerfItem(OperationType.SAVE_ONE, springCost, sdkCost, TIMES));
+        assertPerf(springCost, sdkCost, ACCEPTANCE_DIFF);
```

Comment:
line 83 and 84 seems code pattern except OperationType, looks we can add one method like completeTest to avoiding duplicated code.

Reply:
Updated.